### ISSUE TITLE
Include filter generating multiple queries instead of one

### ISSUE THREAD

**Issue by nivruth-naka** [loopbackio/loopback-next#9345]
### Describe the bug

https://github.com/loopbackio/loopback-next/discussions/9344 

### Logs

```shell
https://github.com/nivruth-naka/lb-include/blob/main/lb-include.log
```

### Reproduction

https://github.com/nivruth-naka/lb-include

**Comment by nivruth-naka** [comment]
@shubhamp-sf ,

getting this error in spite of adding primaryKey: true. any idea?

 _Error: A column called 'id' was added to the attributes of 'pcriskguide' but not marked with 'primaryKey: true'_

**Comment by nivruth-naka** [comment]
Is [this](https://github.com/nivruth-naka/lb-include/blob/main/src/models/main.model.ts#L9)the correct way to add?

**Comment by shubhamp-sf** [comment]
Yes, it is. Although I don't see you using sequelize repository, In your reproduction repo.

**Comment by nivruth-naka** [comment]
thanks @shubhamp-sf . now seeing 

_Error: SequelizeConnectionRefusedError: connect ECONNREFUSED 127.0.0.1:5432_

how does datasource setup connection? is it different from default juggler? I don't have 127.0.0.1:5432 in my code, so unsure where it's picking this from

**Comment by shubhamp-sf** [comment]
It's not different from juggler, but this port is default for postgresql so it might be the case that you're missing out those config in the datasource.

Similar to Below is what works for both juggler and sequelize.

```ts
const config = {
  name: 'db',
  connector: 'postgresql',
  host: 'localhost',
  port: 5003,
  user: 'postgres',
  password: 'super-secret',
  database: 'postgres',
};
```

see if you are missing anything?

**Comment by shubhamp-sf** [comment]
Currently url strings are not respected in the sequelize extension. I've noted down the issue and it'll be fixed in the upcoming releases.
Meanwhile you can use the specific keys to specify settings like `host`, `port`, etc. (they can come from .env)

**Comment by nivruth-naka** [comment]
@shubhamp-sf,
in that case it looks like it's defaulting to postgres local credentials, would be good to throw an error right away. 
if possible, could you please point me to a working example for the connection using host, port, etc?

**Comment by nivruth-naka** [comment]
@shubhamp-sf ,
seeing
_status code 500. Error: SequelizeDatabaseError: operator does not exist: jsonb ~~* unknown_

does this work for postgres jsonb columns as well? if yes, please suggest how to fix the filter


```json
{
  "where": {
    "and": [
      {
        "effectiveDate": {
          "lte": "2023-03-01"
        }
      },
      {
        "expiryDate": {
          "gte": "2023-03-01"
        }
      }
    ]
  },
  "order": "code asc",
  "include": [
    {
      "relation": "trans",
      "scope": {
        "where": {
          "and": [
            {
              "segment": {
                "ilike": "%Ware%"
              }
            }
          ]
        }
      },
      "required": true
    }
  ]
}
```

**Comment by nivruth-naka** [comment]
My use case is basically to include some strings and do a case insensitive search within a `jsonb` column. The above filter (without required = true) works in Loopback / Juggler but is too slow as there are multiple queries.

**Comment by shubhamp-sf** [comment]
> does this work for postgres jsonb columns as well?

This seems to a postgres specific type, and current release of sequelize doesn't address this type. Will raise a PR for this in a day or two and let you know.

Thanks for sharing your use-case.

**Comment by nivruth-naka** [comment]
Thanks @shubhamp-sf ! yes, it is postgres specific but a widely used column type and the current default implementation of Loopback (with Juggler) supports this type of querying.

**Comment by yhrchan** [comment]
I did a quick test and got jsonb column to work with loopback-sequelize

For the model, you have to define it using "object" like so:

  @property({
    type: 'object',
    postgresql: {
      dataType: 'jsonb',
    },
  })
  test?: object;


For the filter, the dot notation could be used where "test" is the jsonb column:

{
  "where": {
  "and":[
    {"test.name": { "ilike": "%an%" }},
    {"test.address.city": { "ilike": "%no%" }}
   ]
  }
}

**Comment by nivruth-naka** [comment]
Thanks @yhrchan , great catch! I'll try this out shortly from my end to see if this helps. However, one thing to keep in mind is this nested format you showed above is different from the `include` syntax used in Juggler/Sequelize for relational object querying and may impact the underlying querying process. But if we are able to use jsonb with this nested syntax and internally have sequelize generate just one join query then it will help us.
cc : @shubhamp-sf

**Comment by shubhamp-sf** [comment]
@nivruth-naka The reason it works with default juggler without specifying property as `object` is that they to do type-casting hack (see the `::TEXT` in your query logs). Which stringifies the jsonb data and perform regular ilike match as It was with text. Which in my opinion isn't utilising jsonb capabilities and might lead to bad performance when the jsonb column contains heavily nested data.

PS: Thanks so much @yhrchan for testing this out.

**Comment by yhrchan** [comment]
@shubhamp-sf do you know how to enable SQL logging outputs from loopback-sequelize?

**Comment by shubhamp-sf** [comment]
@yhrchan `loopback:sequelize:queries` is used for that purpose.

It's mentioned in the debug string ref [here](https://www.npmjs.com/package/@loopback/sequelize#debug-strings-reference).

**Comment by yhrchan** [comment]
@shubhamp-sf thanks got it to work

@nivruth-naka 

{
  "where": {
    "code": {
            "ilike": "%33%"
          }
  },
  "include": [
    {
      "relation": "trans",
      "scope": {
        "where": {
          "test.name": {
            "ilike": "%o%"
          }
        }
      },
      "required": true
    }
  ]
}

**Comment by pookdeveloper** [comment]
Same problem here, and it is not only the way of querying but also that the relationships of the includes do not take them into account in the main table.
Exmaple:
I want to obtain the products including the category where the category name is 'CLOTHING'

From what I see, it launches 2 queries when it should be one, and it shows me both products with category and without category.
